web/wp-content/themes/dctx-v2/inc/acf.php

```diff
@@ -0,0 +1,449 @@
+
+	// Get (Unix) times and convert to integer.
+	$now   = (int) current_time( 'timestamp' );
+	$start = (int) $args['start_date'];
+	$end   = (int) $args['end_date'];
+
+	// No dates? Cool, they're optional.
+	if ( empty( $start ) || empty( $end ) ) {
+		return false;
+	}
+
+	// The block has started, but hasn't expired yet.
+	if ( $start <= $now && $end >= $now ) {
+		return false;
+	}
+
+	// Yes, the block has expired.
+	return true;
+}
+
+/**
+ * Update Layout Titles with Subfield Image and Text Fields
+ *
+ * @author WDS
+ * @param string $block_title Default Field Title.
+ * @param array  $field Field array.
+ * @param string $layout Layout type.
+ * @param int    $i number.
+ *
+ * @url https://support.advancedcustomfields.com/forums/topic/flexible-content-blocks-friendlycustom-collapsed-name/
+ *
+ * @return string new ACF title.
+ */
+function dctx_acf_flexible_content_layout_title( $block_title, $field, $layout, $i ) {
+
+	// Current ACF field name.
+	$current_title = $block_title;
+
+	// Remove layout title from text.
+	$block_heading = '';
+
+	// Set an expired var.
+	$expired = '';
+
+	// Get other options.
+	$other_options = get_sub_field( 'other_options' ) ? get_sub_field( 'other_options' ) : get_field( 'other_options' )['other_options'];
+
+	// Get Background Type.
+	$background = get_sub_field( 'background_options' )['background_type']['value'];
+
+	// If there's no background, just move along...
+	if ( 'none' !== $background ) {
+		$background_repeater = get_sub_field( 'carousel_slides' )[0]['background_options']['background_type']['value'];
+		$background_type     = $background ? $background : $background_repeater;
+
+		$type = dctx_return_flexible_content_layout_value( $background_type );
+
+		// Load image from non-repeater sub field background image, if it exists else Load image from repeater sub field background image, if it exists - Hero.
+		if ( 'image' === $background_type ) {
+			$block_heading .= '<img src="' . esc_url( $type['sizes']['thumbnail'] ) . '" height="30" width="30" class="acf-flexible-title-image" />';
+		}
+
+		if ( 'video' === $background_type ) {
+			$block_heading .= '<div style="font-size: 30px; height: 26px; width: 30px;" class="dashicons dashicons-format-video acf-flexible-title-image"><span class="screen-reader-text">' . esc_html__( 'Video', 'dctx' ) . '</span></div>';
+		}
+	}
+
+	// Set default field title. Don't want to lose this.
+	$block_heading .= '<strong>' . esc_html( $current_title ) . '</strong>';
+
+	// ACF Flexible Content Title Fields.
+	$block_title = get_sub_field( 'title' );
+	$headline    = get_sub_field( 'carousel_slides' )[0]['title'];
+	$text        = $block_title ? $block_title : $headline;
+	$start_date  = $other_options['start_date'];
+	$end_date    = $other_options['end_date'];
+
+	// If the block has expired, add "(expired)" to the title.
+	if ( dctx_has_block_expired(
+			array(
+				'start_date' => $start_date,
+				'end_date'   => $end_date,
+			)
+		)
```

Comment:
:no_entry_sign: **Error**: Multi-line function call not indented correctly; expected 4 spaces but found 8 (*PEAR.Functions.FunctionCallSignature.Indent*).